src/App.tsx

```diff
@@ -1,26 +1,56 @@
+/* eslint-disable max-len */
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+import { getTodos } from './api';
+
+import { useAppSelector } from './hooks/useAppSelector';
+import { todosSlice } from './features/todos';
+
+export const App = () => {
+  const [isLoading, setIsLoading] = useState(false);
+
+  const dispatch = useDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(todosFromApi =>
+        dispatch(todosSlice.actions.loadTodos(todosFromApi)),
+      )
+      .finally(() => setIsLoading(false));
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className="section">
+        <div className="container">
+          <div className="box">
+            <h1 className="title">Todos:</h1>
+
+            <div className="block">
+              <TodoFilter />
+            </div>
+
+            <div className="block">
+              {isLoading && <Loader />}
 
-          <div className="block">
-            <Loader />
-            <TodoList />
+              {!isLoading && todos.length > 0 && <TodoList />}
```

Comment:
The `TodoList` component is rendered without passing the `todos` prop. If `TodoList` expects a `todos` prop, you should pass it like this: `<TodoList todos={todos} />`.